scripts/add_electricity.py

```diff
@@ -298,7 +298,26 @@ def attach_wind_and_solar(n, costs):
             else:
                 capital_cost = costs.at[tech, 'capital_cost']
 
-            n.madd("Generator", ds.indexes['bus'], ' ' + tech,
+            if snakemake.config['electricity']['custom_powerplants']: #adapt "if" in case wind and solar are not part of custom_ppls?
+                p = ppl.query('carrier == @tech')
+                logger.info('Adding {} generators of type {}'.format(len(p), tech))
+                profile = ds['profile'].to_pandas()
+                # rename indices in the following to add them at the correct places in the network:
+                profile = profile.rename(index={profile.index[bus_i]: profile.index[bus_i] + ' ' + tech for bus_i in range(len(profile))})
+                p.bus = p.bus.apply(lambda b: b[:-2] if b[-2:] == '.0' else b) #for some reason buses in custom_ppls may end with .0 -> delete that! .astype(int) does not work?
```

Comment:
Aaah, now I think I get it, that would be tricky. Your `custom_powerplants` file has a `bus` column, hasn't it?

Because, then in `build_powerplants.py`:
```python
def add_custom_powerplants(ppl):
    custom_ppl_query = snakemake.config['electricity']['custom_powerplants']
    if not custom_ppl_query:
        return ppl
    add_ppls = pd.read_csv(snakemake.input.custom_powerplants, index_col=0)
    if isinstance(custom_ppl_query, str):
        add_ppls.query(add_ppls, inplace=True)
    return ppl.append(add_ppls, sort=False)
```
reading in the file will identify this `bus` column as dtype `int` and then when we append will have to extend it to a `float` since it doesn't know what to put for the powerplants in `ppl`.

The fix is to modify it to
```python
def add_custom_powerplants(ppl):
    custom_ppl_query = snakemake.config['electricity']['custom_powerplants']
    if not custom_ppl_query:
        return ppl
    add_ppls = pd.read_csv(snakemake.input.custom_powerplants, index_col=0, dtype={'bus': 'str'})
    if isinstance(custom_ppl_query, str):
        add_ppls.query(custom_ppl_query, inplace=True)
    return ppl.append(add_ppls, sort=False)
```

(Note also that I fixed a second bug in the `add_ppls.query` line).

Can you please add the two fixes to your pull request or another one?